sbin/pfdetect

```diff
@@ -76,85 +78,139 @@ POSIX::sigaction(
 
 my @ORIG_ARGV = @ARGV;
 my %args;
-getopts( 'dhp:', \%args );
+getopts( 'dh:', \%args );
 
 my $daemonize = $args{d};
-my $snortpipe = $args{p};
-my $snortpipe_fh;
 my $portscan_sid = 1200003;
 
-pod2usage( -verbose => 1 ) if ( $args{h} || !$args{p} );
+pod2usage( -verbose => 1 ) if ( $args{h} );
 
 my ($line, $sid, $descr, $priority, $date, $srcmac, $srcip, $dstip);
 
 daemonize($PROGRAM_NAME) if ($daemonize);
+our $PARENT_PID = $$;
 
 $logger->info("initialized");
 
-if ( !open( $snortpipe_fh, '<', "$snortpipe" ) ) {
-    $logger->logdie("unable to open snort pipe ($snortpipe): $!");
-} else {
-    $logger->info("listening on $snortpipe");
+my %CHILDREN;
+my $IS_CHILD = 0;
+my $running = 1;
+
+sub start_detectors {
+    foreach my $id (keys %ConfigDetect) {
+        run_detector($id);
+    }
 }
 
-while (<$snortpipe_fh>) {
-
-    $logger->info("alert received: $_");
-    if ( $_
-        =~ /^(.+?)\s+\[\*\*\]\s+\[\d+:(\d+):\d+\]\s+(.+?)\s+.+?(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})(:\d+){0,1}\s+\-\>\s+(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})(:\d+){0,1}/
-        )
-    {
-        $date  = $1;
-        $sid   = $2;
-        $descr = $3;
-        $srcip = $4;
-        $dstip = $6;
-    } elsif ( $_
-        =~ /^(.+?)\s+\[\*\*\]\s+\[\d+:(\d+):\d+\]\s+Portscan\s+detected\s+from\s+(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/
-        )
-    {
-        $date  = $1;
-        $sid   = $portscan_sid;
-        $srcip = $3;
-        $descr = "PORTSCAN";
-    } elsif ( $_
-        =~ /^(.+?)\[\*\*\] \[\d+:(\d+):\d+\]\s+\(spp_portscan2\) Portscan detected from (\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/
-        )
-    {
-        $date  = $1;
-        $sid   = $portscan_sid;
-        $srcip = $3;
-        $descr = "PORTSCAN";
+=head2 runtask
+
+creates a new child to run a task
```

Comment:
Open to new names if you have some.

It could be :
runDetectomaster
runSpartanParser